--- webkitgtk-2.2.4/Source/WebKit2/GNUmakefile.am.0000	2014-01-20 15:57:54.000000000 +0800
+++ webkitgtk-2.2.4/Source/WebKit2/GNUmakefile.am	2014-02-06 18:45:00.552992105 +0800
@@ -558,6 +558,9 @@
 	-module \
 	-avoid-version
 
+libwebkit2gtkinjectedbundle_la_LIBADD = \
+	libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la
+
 libwebkit2gtkinjectedbundle_la_CPPFLAGS = \
 	-fno-strict-aliasing \
 	-I$(srcdir)/Source/WebKit2/Platform \
--- webkitgtk-2.2.4/Tools/TestWebKitAPI/GNUmakefile.am.0000	2014-01-20 20:48:13.000000000 +0800
+++ webkitgtk-2.2.4/Tools/TestWebKitAPI/GNUmakefile.am	2014-02-06 18:45:40.544118956 +0800
@@ -628,10 +628,10 @@
 
 if ENABLE_WEBKIT2
 noinst_LTLIBRARIES += \
-	Libraries/libTestWebKitAPIInjectedBundle.la
+	Libraries/libTestWebKitAPIInjectedBundle.a
 endif # ENABLE_WEBKIT2
 
-Libraries_libTestWebKitAPIInjectedBundle_la_SOURCES = \
+Libraries_libTestWebKitAPIInjectedBundle_a_SOURCES = \
 	Tools/TestWebKitAPI/InjectedBundleController.cpp \
 	Tools/TestWebKitAPI/InjectedBundleController.h \
 	Tools/TestWebKitAPI/InjectedBundleMain.cpp \
@@ -664,79 +664,77 @@
 	Tools/TestWebKitAPI/gtk/InjectedBundleControllerGtk.cpp \
 	Tools/TestWebKitAPI/gtk/PlatformUtilitiesGtk.cpp
 
-Libraries_libTestWebKitAPIInjectedBundle_la_LDFLAGS = \
+Libraries_libTestWebKitAPIInjectedBundle_a_LDFLAGS = \
 	-rpath ${shell pwd}/$(top_builddir)/Libraries/.libs \
-	$(no_undefined) \
 	-avoid-version \
 	-module
 
-Libraries_libTestWebKitAPIInjectedBundle_la_CPPFLAGS = \
+Libraries_libTestWebKitAPIInjectedBundle_a_CPPFLAGS = \
 	$(Libraries_libTestWebKitAPIMain_la_CPPFLAGS) \
 	-I$(top_builddir)/DerivedSources/InjectedBundle \
 	$(GTK_CFLAGS)
 
-Libraries_libTestWebKitAPIInjectedBundle_la_CXXFLAGS = \
+Libraries_libTestWebKitAPIInjectedBundle_a_CXXFLAGS = \
 	-DGTEST_HAS_RTTI=0 \
 	$(global_cxxflags)
 
-Libraries_libTestWebKitAPIInjectedBundle_la_CFLAGS = \
+Libraries_libTestWebKitAPIInjectedBundle_a_CFLAGS = \
 	$(global_cflags)
 
 
 if ENABLE_WEBKIT2
-noinst_LTLIBRARIES += Libraries/WebExtensions/libWebExtensionTest.la
+noinst_LTLIBRARIES += Libraries/WebExtensions/libWebExtensionTest.a
 endif
 
-Libraries_WebExtensions_libWebExtensionTest_la_SOURCES = \
+Libraries_WebExtensions_libWebExtensionTest_a_SOURCES = \
 	Tools/TestWebKitAPI/Tests/WebKit2Gtk/WebExtensionTest.cpp
 
-Libraries_WebExtensions_libWebExtensionTest_la_LDFLAGS = \
+Libraries_WebExtensions_libWebExtensionTest_a_LDFLAGS = \
 	-rpath ${shell pwd}/$(top_builddir)/Libraries/WebExtensions/.libs \
-	$(no_undefined) \
 	-avoid-version \
 	-module
 
-Libraries_WebExtensions_libWebExtensionTest_la_CPPFLAGS = \
+Libraries_WebExtensions_libWebExtensionTest_a_CPPFLAGS = \
 	-I$(srcdir)/Source/WebKit2/WebProcess/InjectedBundle/API/gtk \
 	-I$(top_builddir)/DerivedSources \
 	-I$(top_builddir)/DerivedSources/WebKit2/webkit2extension/include \
 	-DWEBKIT2_COMPILATION \
 	$(webkit2gtk_tests_cppflags)
 
-Libraries_WebExtensions_libWebExtensionTest_la_CXXFLAGS = \
+Libraries_WebExtensions_libWebExtensionTest_a_CXXFLAGS = \
         $(global_cxxflags)
 
-Libraries_WebExtensions_libWebExtensionTest_la_CFLAGS = \
+Libraries_WebExtensions_libWebExtensionTest_a_CFLAGS = \
         $(global_cflags)
 
 
 if ENABLE_WEBKIT2
-noinst_LTLIBRARIES += Libraries/WebExtensions/libWebProcessTest.la
+noinst_LTLIBRARIES += Libraries/WebExtensions/libWebProcessTest.a
 endif
 
-Libraries_WebExtensions_libWebProcessTest_la_SOURCES = \
+Libraries_WebExtensions_libWebProcessTest_a_SOURCES = \
 	Tools/TestWebKitAPI/Tests/WebKit2Gtk/DOMNodeTest.cpp \
 	Tools/TestWebKitAPI/Tests/WebKit2Gtk/FrameTest.cpp \
 	Tools/TestWebKitAPI/Tests/WebKit2Gtk/WebProcessTest.cpp \
 	Tools/TestWebKitAPI/Tests/WebKit2Gtk/WebProcessTest.h
 
-Libraries_WebExtensions_libWebProcessTest_la_LDFLAGS = \
+Libraries_WebExtensions_libWebProcessTest_a_LDFLAGS = \
 	-rpath ${shell pwd}/$(top_builddir)/Libraries/WebExtensions/.libs \
 	$(no_undefined) \
 	-avoid-version \
 	-module
 
-Libraries_WebExtensions_libWebProcessTest_la_CPPFLAGS = \
+Libraries_WebExtensions_libWebProcessTest_a_CPPFLAGS = \
 	-I$(srcdir)/Source/WebKit2/WebProcess/InjectedBundle/API/gtk \
 	-I$(top_builddir)/DerivedSources \
 	-I$(top_builddir)/DerivedSources/WebKit2/webkit2extension/include \
 	-DWEBKIT2_COMPILATION \
 	$(webkit2gtk_tests_cppflags)
 
-Libraries_WebExtensions_libWebProcessTest_la_CXXFLAGS = \
+Libraries_WebExtensions_libWebProcessTest_a_CXXFLAGS = \
 	$(global_cxxflags)
 
-Libraries_WebExtensions_libWebProcessTest_la_CFLAGS = \
+Libraries_WebExtensions_libWebProcessTest_a_CFLAGS = \
 	$(global_cflags)
 
 
--- webkitgtk-2.2.4/Tools/WebKitTestRunner/GNUmakefile.am.0000	2014-01-20 15:57:54.000000000 +0800
+++ webkitgtk-2.2.4/Tools/WebKitTestRunner/GNUmakefile.am	2014-02-06 18:45:00.552992105 +0800
@@ -117,7 +117,8 @@
 	Tools/WebKitTestRunner/InjectedBundle/TextInputController.h
 
 Libraries_libTestRunnerInjectedBundle_la_LIBADD = \
-	libWebCoreInternals.la
+	libWebCoreInternals.la \
+	libjavascriptcoregtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la libwebkit2gtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la $(GTK_LIBS) -lfontconfig
 
 Libraries_libTestRunnerInjectedBundle_la_LDFLAGS = \
 	-rpath ${shell pwd}/$(top_builddir)/../unix/TestNetscapePlugin/.libs \
